README.md

```diff
@@ -336,6 +336,50 @@ Rails.application.config.app_profiler.backend = AppProfiler::StackprofBackend #
 
 By default, the stackprof backend will be used.
 
+
+## Profile Sampling
+
+The `AppProfiler` middleware can be configured to sample a percentage of requests for profiling. This can be useful for profiling a subset of requests in production without profiling every request.
+
+To enable profile sampling:
+
+```ruby
+AppProfiler.profile_sampler_enabled = true
+# OR
+Rails.application.config.app_profiler.profile_sampler_enabled = true
+```
+
+Apps can disable profile sampling on the fly (killswitch, beta flag etc) via `profile_sampler_status_proc`:
+
+```ruby
+AppProfiler.profile_sampler_status_proc = -> { ... }
+# OR
+Rails.application.config.app_profiler.profile_sampler_status_proc = -> { ... }
+```
+
+Sampling configuration paramets along with their default values
+
+| Parameter                           | Default |
```

Comment:
What about on-CPU?

Reply:
cpu mode got added after the whole refactor